lib/services/local/sessiontracker.go

```diff
@@ -116,16 +118,43 @@ func (s *sessionTracker) GetActiveSessionTrackers(ctx context.Context) ([]types.
 		return nil, trace.Wrap(err)
 	}
 
-	sessions := make([]types.SessionTracker, len(result.Items))
-	for i, item := range result.Items {
+	sessions := make([]types.SessionTracker, 0, len(result.Items))
+
+	// We don't overallocate expired since cleaning up sessions here should be rare.
+	noExpiry := make([]backend.Item, 0)
+	now := time.Now().UTC()
+	for _, item := range result.Items {
 		session, err := unmarshalSession(item.Value)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
 
-		sessions[i] = session
+		// NOTE: This is the session expiry timestamp, not the backend timestamp stored in `item.Expires`.
+		after := session.GetExpires().After(now)
+
+		switch {
+		case after:
+			// Keep any items that aren't expired.
+			sessions = append(sessions, session)
+		case !after && item.Expires.IsZero():
+			// Clear item if expiry is not set on the backend.
+			// We currently don't set the expiry here but we will when #11551 is merged.
+			noExpiry = append(noExpiry, item)
+		default:
+			// If we take this branch, the expiry is set and the backend is responsible for cleaning up the item.
+		}
 	}
 
+	go func() {
```

Comment:
Wrap inside an `if len(noExpiry) > 0`, so most of the time we don't even launch the goroutine?